### PR TITLE
Refactor source code for building the Modbus simulator image

### LINKED ISSUE #133: Refactor Modbus simulator
Refactor source code for building the **edgex-modbus-simulator** docker image.
- use protocol, port as parameters to run Modbus TCP or Modbus RTU simulator

### PR DESCRIPTION
**Pull request by weichou1229**
Use protocol, port as parameters to run Modbus TCP or Modbus RTU simulator.

Fix #133

### CODE REVIEW

**Comment by cloudxxx8** [comment]
please make sure the check can be passed

File: simulator/main.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2018-2020 IOTech Ltd
```
Comment (cloudxxx8) [inline simulator/main.go:3]: it should only be 2020

File: simulator/main.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		server.Close()
+		fmt.Printf("Simulator shutdown. \n")
+	}()
+
+	if *protocol == ProtocolTCP {
+		url := fmt.Sprintf("0.0.0.0:%s", *port)
+		fmt.Printf("Modbus TCP address: %v \n", url)
+
+		if err := server.ListenTCP(url); err != nil {
+			fmt.Printf("%v\n", err)
```
Comment (cloudxxx8) [inline simulator/main.go:49]: add some information in error message

File: simulator/main.go (hunk 1 above)
Comment (weichou1229) [reply simulator/main.go]: @cloudxxx8 Modified.

File: simulator/main.go (hunk 2 above)
Comment (weichou1229) [reply simulator/main.go]: @cloudxxx8 Modified.

**Pull request merged**